### PR TITLE
added tests for truth table generator.

### PR DESCRIPTION
**Pull request by ravikamath123**
added test cases for
- basic operations
   - or
   - and
   - not
   - xor
   - conditional
   - bi conditional
 - complex case
    - (a + ~b) -> b
    - a + ( b * c )

### CODE REVIEW

File: tests/test_logic.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+
+def evaluator3(tb, equation_str, result_str):
+    """Args:
+        tb: truth table
+        equation_str: evalator string to compare against
+        result_str: key in the tb for the result
+    Works only for equation of three variable
+    """
+    for i in range(4):
```
Comment (Vipul-Cariappa) [inline tests/test_logic.py]: I believe the range for this for loop is 8

File: tests/test_logic.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    | 0 | 0 |   1   |
+    | 1 | 0 |   0   |
+    | 0 | 1 |   0   |
+    | 1 | 1 |   1   |
+    """
+
+    eq = a % b
+    table = eq.generate_truth_table()
+
+    evaluator2(table, "int(not((not(a) and b) or (a and (not(b)))))", "a <-> b")
```
Comment (Vipul-Cariappa) [inline tests/test_logic.py]: I guess the 2nd argument for evaluator2 should the following:
`int((not(a) or b) and (not(b) or a))`

File: tests/test_logic.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+
+# test_combinations()
+
+# TODO: remove
+# test_andor()
+# conditional /
+# xor ^
+# biconditioanl %
+# (a + ~b) -> b
```
Comment (Vipul-Cariappa) [inline tests/test_logic.py]: I guess you can remove this

**Review by Vipul-Cariappa — COMMENTED** [review]
Please do the necessary changes

**Comment by Vipul-Cariappa** [comment]
The `test_combinations` is testing for 3 variable equation only. Please write test cases for 1, 2, 4 and 5 variable equations also.
Also test some complex 4 and 5 variable equations (1 each is sufficient). 
Everything else looks great. 👍

**Comment by Vipul-Cariappa** [comment]
Everything looks great 👍.
Please verify the changes I have asked for. Once done I will merge it.

**Comment by ravikamath123** [comment]
I have verified and done the necessary changes.

**Pull request merged**